### PR TITLE
Simplify parser global variables and usage

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -23,7 +23,7 @@
 #include <sysexits.h>
 
 #include "runner.h"
-#include "parse.h"
+#include "parser.h"
```
[the remark this answers is not in the document]
Comment (dburgener) [reply src/main.c:26]: I assume this is to get yyparse_wrapper in a header?  Would declaring it somewhere else, such as parse_functions.h work?

File: src/main.c [hunk 2]
```diff
@@ -36,10 +36,6 @@
 #define COLOR_ID        129
 #define SUMMARY_ONLY_ID 130
 
-extern int yydebug;
```
Comment (dburgener) [inline src/main.c:39]: I don't think this needs to move to a header.  It's only used here and in the generated parse.{c,h}.

**Comment by cgzones** [comment]
The main point is to get rid of the global variables `ast` and `cur`.
* currently before any first syntax tree operation `if(!cur) { cur = ast; }` must be called
* not setting `parsing_filename` might lead to wrong messages or worse
* `cur` is just a state pointer for the parses, not designated for the outside

I think this leads to cleaner code and is more maintainable especially if in the future the parser gets extended to parse `access_vectors` files or the like.

In the recent update I dropped the former introduced file `parser.h` and included the prototype into the bison generated header file.

**Comment by dburgener** [comment]
Great, love the general idea, and things look a lot more reasonable with the change to put yyparse_wrapper in parse.{c,h}.  I'll go through all this in detail as soon as I get a chance.

**Comment by dburgener** [comment]
This is wonderful!  Thanks!

**Pull request merged**